src/components/PeoplePage.tsx

```diff
@@ -1,33 +1,92 @@
-import { PeopleFilters } from './PeopleFilters';
+import { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Person } from '../types';
 import { PeopleTable } from './PeopleTable';
+import { Outlet, useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+  const [data, setData] = useState<Person[] | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [searchParams] = useSearchParams();
 
-      <div className="block">
-        <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+  useEffect(() => {
+    setLoading(true);
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+    fetch('./api/people.json')
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Помилка завантаження JSON');
+        }
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+        return response.json();
+      })
+      .then(people => {
+        let filteredPeople: Person[] = people;
 
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+        const query = searchParams.get('query')?.toLowerCase() || '';
+        const sex = searchParams.get('sex');
+        const centuries = searchParams.getAll('centuries');
+
+        if (query) {
+          filteredPeople = filteredPeople.filter(person =>
+            person.name.toLowerCase().includes(query),
+          );
+        }
+
+        if (sex) {
+          filteredPeople = filteredPeople.filter(person => person.sex === sex);
+        }
+
+        if (centuries.length > 0) {
+          filteredPeople = filteredPeople.filter(person => {
+            const birthCentury = Math.ceil(person.born / 100).toString();
+            const deathCentury = Math.ceil(person.died / 100).toString();
+
+            return (
+              centuries.includes(birthCentury) ||
+              centuries.includes(deathCentury)
+            );
+          });
+        }
 
-              <p>There are no people matching the current search criteria</p>
+        setData(filteredPeople);
+        setError(null);
+      })
+      .catch(() => {
+        setError('Something went wrong');
+        setData([]);
+      })
+      .finally(() => setTimeout(() => setLoading(false), 100));
+  }, [searchParams]);
```

Comment:
problem with tests looks like here in this useEffect.
After every change search params occurs fetch data, because search params in array dependencies.
Fetching people should be once (put after fetch people list in state) and than just filter existing list.